great_expectations/checkpoint/actions.py

```diff
@@ -300,6 +300,73 @@ def _root_validate_slack_params(cls, values: dict) -> dict:
 
         return values
 
+    @override
+    def v1_run(
+        self, checkpoint_result: CheckpointResult, action_context: ActionContext | None = None
+    ) -> dict:
+        success = checkpoint_result.success or False
+        result = {"slack_notification_result": "none required"}
+        if not self._is_enabled(success=success):
+            return result
+
+        checkpoint_text_blocks: list[dict] = []
+        for (
+            validation_result_suite_identifier,
+            validation_result_suite,
+        ) in checkpoint_result.run_results.items():
+            validation_text_blocks = self._render_validation_result(
+                result_identifier=validation_result_suite_identifier,
+                result=validation_result_suite,
+                action_context=action_context,
+            )
+            checkpoint_text_blocks.extend(validation_text_blocks)
+
+        query = self.renderer.build_query(
+            checkpoint_result=checkpoint_result, text_blocks=checkpoint_text_blocks
+        )
+        blocks = query.get("blocks")
+        if blocks:
+            if len(blocks) >= 1:
+                if blocks[0].get("text"):
+                    result = self._send_notifications_in_batches(
+                        blocks=blocks, query=query, result=result
+                    )
+                else:
+                    result = self._get_slack_result(query=query)
```

Comment:
There's a decent bit of duplication from the existing run method but I found it easier to do this than refactor old code that we're going to soon delete?

Reply:
yeah, seems legit to me.